### PR TITLE
feat(menu): Add --selected class to menu items

### PR DESCRIPTION
**Pull request by williamernest**
fixes: #2031 
related: #536
Adds `mdc-list-item--selected` class to menu items when `rememberSelection` is set to `true`. 
Auto-focuses on the last selected item when opened when `rememberSelection` is set to `true`.

### CODE REVIEW

File: packages/mdc-menu/adapter.js [hunk 1]
```diff
@@ -141,6 +141,34 @@ class MDCMenuAdapter {
 
   /** @param {string} height */
   setMaxHeight(height) {}
+
+  /** @param {number} index */
+  getOptionAtIndex(index) {}
```
Comment (lynnjepsen) [inline packages/mdc-menu/adapter.js]: does this return something? It is a getter after all...

File: packages/mdc-menu/constants.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ const cssClasses = {
   ANIMATING_OPEN: 'mdc-menu--animating-open',
   ANIMATING_CLOSED: 'mdc-menu--animating-closed',
   LIST_ITEM: 'mdc-list-item',
+  SELECTED_LIST_ITEM: 'mdc-list-item--selected',
```
Comment (lynnjepsen) [inline packages/mdc-menu/constants.js]: you should be able to reference the constants on MDCListFoundation.strings

File: packages/mdc-menu/foundation.js [hunk 3]
```diff
@@ -137,6 +142,12 @@ class MDCMenuFoundation extends MDCFoundation {
     this.anchorMargin_ = {top: 0, right: 0, bottom: 0, left: 0};
     /** @private {?AutoLayoutMeasurements} */
     this.measures_ = null;
+    /** @private {number} */
+    this.selectedIndex_ = -1;
+    /** @private {boolean} */
+    this.rememberSelection_ = false;
+    /** @private {boolean} */
+    this.keyDownWithinMenu_ = false;
```
Comment (lynnjepsen) [inline packages/mdc-menu/foundation.js]: this keeps track of whether the user *currently* has a keydown interaction happening on the `mdc-menu` component?

These sorts of fields can be vague...so add a comment about exactly what it is this field represents

File: packages/mdc-menu/foundation.js [hunk 4]
```diff
@@ -188,13 +199,24 @@ class MDCMenuFoundation extends MDCFoundation {
...
+  }
+
   /**
    * @param {?number} focusIndex
    * @private
    */
   focusOnOpen_(focusIndex) {
     if (focusIndex === null) {
-      // First, try focusing the menu.
+      // First focus the last selected item if we're remembering selections and it has been set
```
Comment (lynnjepsen) [inline packages/mdc-menu/foundation.js]: flip this
"If this instance of MDCMenu remembers selections, and the user has made a selection, then focus the last selected item"

File: packages/mdc-menu/foundation.js [hunk 5]
```diff
@@ -241,6 +263,9 @@ class MDCMenuFoundation extends MDCFoundation {
     const isArrowUp = key === 'ArrowUp' || keyCode === 38;
     const isArrowDown = key === 'ArrowDown' || keyCode === 40;
     const isSpace = key === 'Space' || keyCode === 32;
+    const isEnter = key === 'Enter' || keyCode === 13;
+
+    this.keyDownWithinMenu_ = isEnter || isSpace;
```
Comment (lynnjepsen) [inline packages/mdc-menu/foundation.js]: explain whats going on here

File: packages/mdc-menu/foundation.js [hunk 6]
```diff
@@ -188,13 +199,24 @@ class MDCMenuFoundation extends MDCFoundation {
...
    */
   focusOnOpen_(focusIndex) {
     if (focusIndex === null) {
-      // First, try focusing the menu.
+      // First focus the last selected item if we're remembering selections and it has been set
+      if (this.rememberSelection_ && this.selectedIndex_ >= 0) {
+        this.adapter_.focusItemAtIndex(this.selectedIndex_);
+        return;
+      }
+      // Then, try focusing the menu.
```
Comment (lynnjepsen) [inline packages/mdc-menu/foundation.js]: You probably dont need this comment.

File: packages/mdc-menu/foundation.js [hunk 7]
```diff
@@ -297,7 +322,12 @@ class MDCMenuFoundation extends MDCFoundation {
     const isEscape = key === 'Escape' || keyCode === 27;
 
     if (isEnter || isSpace) {
-      this.handlePossibleSelected_(evt);
+      // If our menu didn't catch the keydown, then the user likely hit the
```
Comment (lynnjepsen) [inline packages/mdc-menu/foundation.js]: There is no "we" or "our" in comments. The program is the thing running. Try and be specific about which part of the program is doing what action.

For example,

"If MDCMenu does not see a keydown event, then it assumes the user triggered an enter or space event"

Also, how is enter and space NOT a keydown event??

File: packages/mdc-menu/foundation.js [hunk 8]
```diff
@@ -564,6 +597,38 @@ class MDCMenuFoundation extends MDCFoundation {
   isOpen() {
     return this.isOpen_;
   }
+
+  /** @return {?Element} */
```
Comment (lynnjepsen) [inline packages/mdc-menu/foundation.js]: I think we try NOT to reference HTML Element in our Foundation classes. Maybe this public method is not necessary...

File: packages/mdc-menu/constants.js (hunk 2 above)
Comment (williamernest) [reply packages/mdc-menu/constants.js]: MDCListFoundation doesn't exist

**Pull request merged**